low-level/solution/lr2/server/health.py

```diff
@@ -161,11 +161,11 @@ def set_health_data(health_data: dict, status, description=None,
                         recommendation=None, specifics=None):
         """Sets health attributes for a component."""
         good_state = (status == "OK")
-        if not description:
+        if description in ["NA", "N/A", None, ""]:
```

Comment:
Is there any possibility in which state is faulty (or anything other than OK) and description comes as NA?
Because in that case, description will mention good health even if state is not OK.

Reply:
No, If health is not ok 
description will be "%s is not in good health"

Reply:
As Sandeep said.

Reply:
For some resources, like storage PSUs, we are getting description directly from resource rather than building our own in `set_health_data()`. 
What I am trying to say is, for such resources, can there be a case possible in which health status is other than `OK` and description is `NA`
Because in that case below code will set the description mentioning good health even if heath is not `OK`
```
if description in ["NA", "N/A", None, ""]:
    description = "%s %s in good health." % (health_data.get("uid"),'is' if good_state else 'is not')
```

Reply:
`can there be a case possible in which health status is other than OK and description is NA`
If description is NA, the description will be set based on Health status.

`Because in that case below code will set the description mentioning good health even if heath status is other than OK` 
No, `'is' if good_state else 'is not'` changes description.

Reply:
Got it. Thanks for the explanation @mariyappanp @sandeepsgt